internal/plugin/plugin.go

```diff
@@ -147,6 +147,9 @@ func Load(path string, basedir string) (*Plugin, error) {
 // FilterByPluginsEnv returns matching plugins based on a space separated list of plugins (and optional versions) to include.
 func FilterByPluginsEnv(plugins []*Plugin, pluginsEnv string) ([]*Plugin, error) {
 	if pluginsEnv == "" {
+		return nil, nil
+	}
```

Comment:
Made these utilities more strict so if no filter is provided, they will default to not building anything.